src/shims/unix/freebsd/sync.rs

```
@@ -0,0 +1,216 @@
+                };
+                let futex_ref = futex_ref.futex.clone();
+
+                let count = match val{
+                    u64::MAX => usize::MAX, // Preserve MAX because it specifies to wake everyone.
+                    // This fits because we did `read_target_usize`.
+                    val => usize::try_from(val).expect("`futex_wait` accepts `count` as usize, which can't seem to fit `val` of `_umtx_op`")
+                };
+
+                // Read the Linux futex implementation in Miri to understand why this fence is needed.
+                this.atomic_fence(AtomicFenceOrd::SeqCst)?;
+                // `_umtx_op` doesn't return the amount of woken threads.
+                let _woken = this.futex_wake(
+                    &futex_ref,
+                    u32::MAX, // we set the bitset to include all bits
+                    count,
+                )?;
+                this.write_int(0, dest)?;
+                interp_ok(())
+            }
+            op => {
+                throw_unsup_format!("Miri does not support `_umtx_op` syscall with op={}", op)
+            }
+        }
+    }
+
+    /// Parses a `_umtx_time` struct.
+    /// Returns `None` if the underlying `timespec` struct is invalid.
+    fn read_umtx_time(&mut self, ut: &MPlaceTy<'tcx>) -> InterpResult<'tcx, Option<UmtxTime>> {
+        let this = self.eval_context_mut();
+        let timespec_place = this.project_field(ut, 0)?;
+        // Inner `timespec` must still be valid.
+        let duration = match this.read_timespec(&timespec_place)? {
+            Some(dur) => dur,
+            None => return interp_ok(None),
+        };
+        let flags_place = this.project_field(ut, 1)?;
+        let flags = this.read_scalar(&flags_place)?.to_u32()?;
+        let clock_id_place = this.project_field(ut, 2)?;
+        let clock_id = this.read_scalar(&clock_id_place)?.to_u32()?;
+        interp_ok(Some(UmtxTime { timeout: duration, flags, _clock_id: clock_id }))
```

Please fully interpret the struct data here:
- Instead of `flags`, have a field `abstime: bool` in the struct. `flags` must be either 0 or UMTX_ABSTIME, all other values are unsupported.
- Interpret the clock ID. This will encode whether to use the system clock or monotonic clock. The logic should probably be the same as [here](https://github.com/rust-lang/rust/blob/562b282fcc8b6ffa3770d30cd40df847ebb56182/src/tools/miri/src/shims/unix/sync.rs#L327-L339).

---

Ah, that's the clock ID. I got it!

FreeBSD has a lot of clocks:
`CLOCK_REALTIME` but also `CLOCK_REALTIME_FAST`,...
`CLOCK_MONOTONIC` but also `CLOCK_MONOTONIC_FAST`, ...
And a lot more, so I should only support the above 2, but what about its variants? Just interpret them as the default, or don't support them?

---

The proper thing to do is to share the code with [here](https://github.com/rust-lang/miri/blob/9611e78d8a9ba3eae6b3741edccf2c2d63e4eb42/src/shims/time.rs#L39-L76) and have a type like
```rust
enum ClockKind { System, Monotonic }
```
and a function that converts a clock_id to that enum, and use it everywhere. Then we can centrally decide which of the variants it makes sense to support.

But I'm fine with postponing this, and just supporting CLOCK_REALTIME and CLOCK_MONOTONIC for now.

Any other, unknown clock value should throw an "unsupported" error.